libafl/src/events/mod.rs

```diff
@@ -64,6 +65,15 @@ pub static mut EVENTMGR_SIGHANDLER_STATE: ShutdownSignalData = ShutdownSignalDat
 #[derive(Debug, Clone)]
 pub struct ShutdownSignalData {
     shutting_down: bool,
+    shmem_allocated: bool,
+}
+
+#[cfg(all(unix, feature = "std"))]
+impl ShutdownSignalData {
+    /// Set the flag to true, indicating that this process has allocated shmem
+    pub fn set_shmem_allocated(&mut self) {
+        self.shmem_allocated = true;
```

Comment:
Does this write need to be written volatile?

Reply:
maybe

Reply:
made it volatile